k9mail/src/main/java/com/fsck/k9/mailstore/LocalStore.java

```diff
@@ -789,6 +789,13 @@ private static Part findPartById(Part searchRoot, long partId) {
 
             if (part instanceof LocalMessage) {
                 LocalMessage localMessage = (LocalMessage) part;
+                if (localMessage.getMessagePartId() == partId) {
+                    return localMessage;
+                }
+            }
+
+            if (part instanceof LocalMimeMessage) {
```

Comment:
You can now check for `LocalPart` instead of `LocalBodyPart` and `LocalMimeMessage`.